src/rmplan/agent.ts

```diff
@@ -41,32 +57,49 @@ export async function rmplanAgent(planFile: string, options: any, globalCliOptio
   // Determine the base directory for operations
   let currentBaseDir = await getGitRoot();
 
-  // Handle workspace creation if a task ID is provided
-  if (options.workspace) {
-    log(`Workspace task ID provided: ${options.workspace}`);
+  // Handle workspace creation or auto-selection
+  if (options.workspace || options.autoWorkspace) {
+    const workspaceManager = new WorkspaceManager(currentBaseDir);
+    let workspace;
+    let selectedWorkspace;
 
-    // Verify the original plan file exists
-    try {
-      // Use stat to check if file exists
-      try {
-        await Bun.file(currentPlanFile).text();
-      } catch {
-        error(`Original plan file ${currentPlanFile} does not exist or is empty.`);
-        process.exit(1);
+    if (options.autoWorkspace) {
+      // Use auto-selector to find or create a workspace
+      log('Auto-selecting workspace...');
+      const selector = new WorkspaceAutoSelector(workspaceManager, config);
+      const taskId = options.workspace || `auto-${Date.now()}`;
+
+      selectedWorkspace = await selector.selectWorkspace(taskId, currentPlanFile, {
+        interactive: !options.nonInteractive,
+        preferNewWorkspace: options.preferNewWorkspace,
+      });
+
+      if (selectedWorkspace) {
+        workspace = {
+          path: selectedWorkspace.workspace.workspacePath,
+          originalPlanFilePath: selectedWorkspace.workspace.originalPlanFilePath,
+          taskId: selectedWorkspace.workspace.taskId,
+        };
+
+        if (selectedWorkspace.isNew) {
+          log(`Created new workspace for task: ${workspace.taskId}`);
+        } else {
+          log(`Selected existing workspace for task: ${selectedWorkspace.workspace.taskId}`);
+          if (selectedWorkspace.clearedStaleLock) {
+            log('(Cleared stale lock)');
+          }
+        }
       }
-    } catch (err) {
-      error(`Error checking original plan file: ${String(err)}`);
-      process.exit(1);
+    } else {
+      // Create a workspace with the specified task ID
+      log(`Creating workspace for task: ${options.workspace}`);
+      workspace = await workspaceManager.createWorkspace(
+        options.workspace,
+        currentPlanFile,
+        config
+      );
```

Comment:
Here, first see if the workspace exists already. If so then use it. If not, then create it if newWorkspace is true, and otherwise throw an error.